vignettes/finalize.Rmd

```diff
@@ -0,0 +1,162 @@
+---
+title: "Finalizing and exporting CMAP-themed plots"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Finalizing and exporting CMAP-themed plots}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>",
+  fig.width = 7,
+  fig.asp = 400/670,
+  fig.retina = 4,
+  fig.align = "center"
+)
+
+library(tidyverse)
+library(cmapplot)
+
+# clean up dataset
+df <- transit_ridership %>% 
+  filter(system != "pace_ada") %>% 
+  mutate(system = recode_factor(system,
+    cta_bus = "CTA bus",
+    cta_rail = "CTA rail",
+    metra = "Metra",
+    pace = "Pace"
+  ))
+
+# build plot
+p <- ggplot(data = df, 
+            mapping = aes(x = year, y = ridership, color = system)) +
+  geom_recessions(ymin = 0) +
+  geom_line() +
+  geom_text_lastonly(
+    mapping = aes(label = round(ridership, digits = 0)), 
+    add_points = TRUE,
+    nudge_x = 0.5) +
+  coord_cartesian(clip = "off") + 
+  theme_cmap()
+```
+
+`finalize_plot()` will place a ggplot into a frame defined by CMAP design standards. It will align your title and caption to the left, add a horizontal line on top, and make other adjustments. It can show you the final plot and/or export it as a raster or vector file. This function will not apply CMAP design standards to the plot itself; use `theme_cmap()` for that.
+
+The function has a number of customization options built in, accepting 16 (or more) arguments. Refer to the object documentation, `?finalize_plot`, for detailed information on all arguments.
+
+In this vignette we will use the final version of the line chart developed in `vignette("plots")`. That plot is referenced in the following examples as `p`.
+
+## Basic implementation
+
+After creating a plot and applying `theme_cmap()`, use `finalize_plot()` to complete the implementation of
+CMAP design standards. You will probably want to set at least the `title` and `caption`, although the function will extract them from the ggplot if they were specified. As you are preparing the plot, you will likely want to view it within R. Do this by leaving leaving the default `mode = "plot"` to send the finished plot to the "Plots" tab within RStudio. Windows users may prefer `mode = "window"`, which draws the plot in a popup window. In both modes, the plot will show up "actual size" (depending on your screen's resolution) surrounded by a grey canvas.
+
+```{r finalize1, fig.width= 11, out.width = "100%"}
```

Comment:
@nmpeterson I found a way to scale outputs of `finalize_plot` without relying on png export and re-placement. I've used it here to demonstrate the canvas drawn around the plot in R. I wanted to flag it for you in case you think it's worth implementing elsewhere? I believe this can be set globally at the top of the file.

Reply:
This is a great adjustment. I've implemented everywhere except the one plot in `finalize` that is explicitly an example about using the `mode` and `filename` parameters.